cpp/include/cudf/rolling.hpp

```diff
@@ -144,6 +147,16 @@ struct window_bounds {
   size_type const _value;    ///< Finite window boundary value (in days or rows)
 };
 
+/**
+ * @brief Indicates which endpoints a rolling window contains.
+ */
+enum class window_type : int32_t {
```

Comment:
One wonders if we might use a more descriptive name for this type.  Something like `window_margin(_type)`?